packages/js/product-editor/src/blocks/generic/linked-product-list/edit.tsx

```diff
@@ -68,73 +70,125 @@ export function EmptyStateImage( {
 	}
 }
 
+async function searchProducts(
+	searchValue = '',
+	excludedIds: number[] = []
+): Promise< Product[] > {
+	return resolveSelect( PRODUCTS_STORE_NAME ).getProducts< Product[] >( {
+		search: searchValue,
+		orderby: 'title',
+		order: 'asc',
+		per_page: 5,
+		exclude: excludedIds,
+	} );
+}
+
 export function LinkedProductListBlockEdit( {
 	attributes,
 	context: { postType, isInSelectedTab },
 }: ProductEditorBlockEditProps< LinkedProductListBlockAttributes > ) {
 	const { property, emptyState } = attributes;
+	const loadInitialSearchResults = useRef( false );
+	const [ , setSearchValue ] = useState( '' );
+	const [ searchedProducts, setSearchedProducts ] = useState< Product[] >(
+		[]
+	);
+	const [ isSearching, setIsSearching ] = useState( false );
 	const blockProps = useWooBlockProps( attributes );
 	const [ state, dispatch ] = useReducer( reducer, {
 		linkedProducts: [],
-		searchedProducts: [],
 	} );
 
 	const productId = useEntityId( 'postType', postType );
 
 	const loadLinkedProductsDispatcher =
 		getLoadLinkedProductsDispatcher( dispatch );
-	const searchProductsDispatcher = getSearchProductsDispatcher( dispatch );
 	const selectSearchedProductDispatcher =
 		getSelectSearchedProductDispatcher( dispatch );
 	const removeLinkedProductDispatcher =
 		getRemoveLinkedProductDispatcher( dispatch );
-
 	const [ linkedProductIds, setLinkedProductIds ] = useProductEntityProp<
 		number[]
 	>( property, { postType } );
 
 	useEffect( () => {
-		if ( ! state.selectedProduct ) {
+		if (
+			! state.selectedProduct &&
+			linkedProductIds &&
```

Comment:
With this check here, we won't need the logical operator a couple of lines below
```
loadLinkedProductsDispatcher( linkedProductIds ?? [] );
```